Source/playerinfo/PlayerInfo.cpp

```diff
@@ -0,0 +1,406 @@
+                value = false;
+            }
+        }
+        return value;
+    }
+
+    Exchange::IPlayerProperties::PlaybackResolution PlaybackResolution() const
+    {
+        Exchange::IPlayerProperties::PlaybackResolution value = Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_UNKNOWN;
+        if (_playerConnection != nullptr) {
+            if (_playerConnection->Resolution(value) != Core::ERROR_NONE) {
+                value = Exchange::IPlayerProperties::PlaybackResolution::RESOLUTION_UNKNOWN;
+            }
+        }
+        return value;
+    }
+
+    int8_t VideoCodecs(playerinfo_videocodec_t array[], const uint8_t length) const
+    {
+        Exchange::IPlayerProperties::IVideoCodecIterator* videoCodecs;
+        int8_t value = 0;
+
+        if (_playerConnection != nullptr) {
+
+            if (_playerConnection->VideoCodecs(videoCodecs) == Core::ERROR_NONE && videoCodecs != nullptr) {
+
+                Exchange::IPlayerProperties::VideoCodec codec;
+
+                uint8_t numberOfCodecs = 0;
+                playerinfo_videocodec_t newArray[50];
+
+                while (videoCodecs->Next(codec) && numberOfCodecs < sizeof(newArray)) {
+                    switch (codec) {
+                    case Exchange::IPlayerProperties::VideoCodec::VIDEO_UNDEFINED:
+                        newArray[numberOfCodecs] = PLAYERINFO_VIDEO_UNDEFINED;
+                        break;
+                    case Exchange::IPlayerProperties::VideoCodec::VIDEO_H263:
+                        newArray[numberOfCodecs] = PLAYERINFO_VIDEO_H263;
+                        break;
+                    case Exchange::IPlayerProperties::VideoCodec::VIDEO_H264:
+                        newArray[numberOfCodecs] = PLAYERINFO_VIDEO_H264;
+                        break;
+                    case Exchange::IPlayerProperties::VideoCodec::VIDEO_H265:
+                        newArray[numberOfCodecs] = PLAYERINFO_VIDEO_H265;
+                        break;
+                    case Exchange::IPlayerProperties::VideoCodec::VIDEO_H265_10:
+                        newArray[numberOfCodecs] = PLAYERINFO_VIDEO_H265_10;
+                        break;
+                    case Exchange::IPlayerProperties::VideoCodec::VIDEO_MPEG:
+                        newArray[numberOfCodecs] = PLAYERINFO_VIDEO_MPEG;
+                        break;
+                    case Exchange::IPlayerProperties::VideoCodec::VIDEO_VP8:
+                        newArray[numberOfCodecs] = PLAYERINFO_VIDEO_VP8;
+                        break;
+                    case Exchange::IPlayerProperties::VideoCodec::VIDEO_VP9:
+                        newArray[numberOfCodecs] = PLAYERINFO_VIDEO_VP9;
+                        break;
+                    case Exchange::IPlayerProperties::VideoCodec::VIDEO_VP10:
+                        newArray[numberOfCodecs] = PLAYERINFO_VIDEO_VP10;
+                        break;
+                    }
```

Comment:
If the enum is not found the array element will contain garbage, as the counter is still incremented.